extensions/amp-story/1.0/parallax-fx.js

```diff
@@ -0,0 +1,352 @@
+   * change event then update each of the layers will its calculated position.
+   * @param {Event} event
+   * @param {!Array<!ParallaxPage>} pages
+   * @private
+   */
+  parallaxOrientationMutate_(event, pages) {
+    const window = this.win_;
+    const {screen} = window;
+    let {gamma, beta} = event;
+
+    let angle;
+
+    // Detect the implementation of orientation angle
+    if ('orientation' in screen) {
+      angle = screen.orientation.angle;
+    } else {
+      angle = window.orientation;
+    }
+
+    // Reverse gamma/beta if the device is in landscape
+    if (window.orientation == 90 || window.orientation == -90) {
+      const tmp = gamma;
+      gamma = beta;
+      beta = tmp;
+    }
+
+    // Flip signs of the angles if the phone is in 'reverse landscape' or
+    // 'reverse portrait'
+    if (angle < 0) {
+      gamma = -gamma;
+      beta = -beta;
+    }
+
+    // Smooth the gamma value (X-AXIS)
+    if (this.smoothingPointsX_.length > SMOOTHING_PTS) {
+      this.smoothingPointsX_.shift();
+    }
+    this.smoothingPointsX_.push(gamma);
+    const avgX = sum(this.smoothingPointsX_) / SMOOTHING_PTS;
+    if (this.smoothingPointsX_.length > SMOOTHING_PTS && this.middleX_ == 0) {
+      this.middleX_ = avgX;
+    }
+
+    // Smooth the beta value (Y-AXIS)
+    if (this.smoothingPointsY_.length > SMOOTHING_PTS) {
+      this.smoothingPointsY_.shift();
+    }
+    this.smoothingPointsY_.push(beta);
+    const avgY = sum(this.smoothingPointsY_) / SMOOTHING_PTS;
+    if (this.smoothingPointsY_.length > SMOOTHING_PTS && this.middleY_ == 0) {
+      this.middleY_ = avgY;
+    }
+
+    let mappedX = avgX - this.middleX_;
+    let mappedY = avgY - this.middleY_;
+
+    // Limit the range for a smoother/less shaky effect
+    mappedX = mapRange(mappedX, -75, 75, -25, 25);
+    mappedY = mapRange(mappedY, -75, 75, -25, 25);
+
+    pages.forEach(page => {
+      if (page.shouldUpdate()) {
+        if (this.middleY_ != 0 && this.middleX_ != 0) {
+          page.update(mappedX, mappedY);
+        } else {
+          page.update(this.middleX_, this.middleY_);
+        }
+      }
+    });
+  }
+}
+
+/**
+ * Encapsulates and tracks a story page's parallax effect.
+ */
+export class ParallaxPage {
+  /**
+   * @param {!Element} page The parent page of thi element
+   * @param {!../../../src/service/vsync-impl.Vsync} vsync
+   */
+  constructor(page, vsync) {
+    /** @private @const {!Element} */
+    this.page_ = page;
+
+    /** @private @const {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = vsync;
+  }
+
+  /**
+   * Apply the parallax effect to the element given the movements in the
+   * x and y axes
+   * @param {number} x The movement of the layer in the x axis
+   * @param {number} y The movement of the layer in the y axis
+   */
+  update(x = 0, y = 0) {
+    this.vsync_.mutate(() => {
+      setStyles(this.page_, {
+        perspectiveOrigin: `${Math.round(x * 50)}px ${Math.round(y * 50)}px`,
```

Comment:
Where does `50` come from? This should be a document constant at the top of the file.